diff --git a/libqalculate/util.cc b/libqalculate/util.cc
index 418dfe4..7996834 100644
--- a/libqalculate/util.cc
+++ b/libqalculate/util.cc
@@ -455,48 +455,13 @@ string getOldLocalDir() {
 #endif
 }
 string getLocalDir() {
-#ifdef _WIN32
-	char path[MAX_PATH];
-	SHGetFolderPathA(NULL, CSIDL_LOCAL_APPDATA | CSIDL_FLAG_CREATE, NULL, 0, path);
-	string str = path;
-	return str + "\\Qalculate";
-#else
-	const char *homedir;
-	if((homedir = getenv("XDG_CONFIG_HOME")) == NULL) {
-		return string(getpwuid(getuid())->pw_dir) + "/.config/qalculate";
-	}
-	return string(homedir) + "/qalculate";
-#endif
+	return string(getenv("SNAP_USER_DATA"));
 }
 string getLocalDataDir() {
-#ifdef _WIN32
-	char path[MAX_PATH];
-	SHGetFolderPathA(NULL, CSIDL_LOCAL_APPDATA | CSIDL_FLAG_CREATE, NULL, 0, path);
-	string str = path;
-	return str + "\\Qalculate";
-#else
-	const char *homedir;
-	if((homedir = getenv("XDG_DATA_HOME")) == NULL) {
-		return string(getpwuid(getuid())->pw_dir) + "/.local/share/qalculate";
-	}
-	return string(homedir) + "/qalculate";
-#endif
+	return string(getenv("SNAP_USER_COMMON"));
 }
 string getLocalTmpDir() {
-#ifdef _WIN32
-	char path[MAX_PATH];
-	SHGetFolderPathA(NULL, CSIDL_LOCAL_APPDATA | CSIDL_FLAG_CREATE, NULL, 0, path);
-	string str = path;
-	str += "\\cache";
-	_mkdir(str.c_str());
-	return str + "\\Qalculate";
-#else
-	const char *homedir;
-	if((homedir = getenv("XDG_CACHE_HOME")) == NULL) {
-		return string(getpwuid(getuid())->pw_dir) + "/.cache/qalculate";
-	}
-	return string(homedir) + "/qalculate";
-#endif
+	return string(getenv("SNAP_USER_DATA"));
 }
 
 bool move_file(const char *from_file, const char *to_file) {
@@ -536,69 +501,18 @@ bool move_file(const char *from_file, const char *to_file) {
 }
 
 string getPackageDataDir() {
-#ifndef WIN32
-	return PACKAGE_DATA_DIR;
-#else
-	char exepath[MAX_PATH];
-	GetModuleFileName(NULL, exepath, MAX_PATH);
-	string datadir(exepath);
-	datadir.resize(datadir.find_last_of('\\'));
-	if (datadir.substr(datadir.length() - 4) == "\\bin") {
-		datadir.resize(datadir.find_last_of('\\'));
-		datadir += "\\share";
-	} else if(datadir.substr(datadir.length() - 6) == "\\.libs") {
-		datadir.resize(datadir.find_last_of('\\'));
-		datadir.resize(datadir.find_last_of('\\'));
-		return datadir;
-	}
-	return datadir;
-#endif
+	const char *snap = getenv("SNAP");
+	string datadir(snap);
+	datadir += "/share";
+ 	return datadir;
 }
 
 string getGlobalDefinitionsDir() {
-#ifndef WIN32
-	return string(PACKAGE_DATA_DIR) + "/qalculate";
-#else
-	char exepath[MAX_PATH];
-	GetModuleFileName(NULL, exepath, MAX_PATH);
-	string datadir(exepath);
-	bool is_qalc = datadir.substr(datadir.length() - 8) == "qalc.exe";
-	datadir.resize(datadir.find_last_of('\\'));
-	if(datadir.substr(datadir.length() - 4) == "\\bin") {
-		datadir.resize(datadir.find_last_of('\\'));
-		datadir += "\\share\\qalculate";
-		return datadir;
-	} else if(datadir.substr(datadir.length() - 6) == "\\.libs") {
-		datadir.resize(datadir.find_last_of('\\'));
-		datadir.resize(datadir.find_last_of('\\'));
-		if(!is_qalc) {
-			datadir.resize(datadir.find_last_of('\\'));
-			datadir += "\\libqalculate";
-			if(!dirExists(datadir)) {
-				datadir += "-";
-				datadir += VERSION;
-			}
-		}
-		return datadir + "\\data";
-	}
-	return datadir + "\\definitions";
-#endif
+	return getPackageDataDir() + "/qalculate";
 }
 
 string getPackageLocaleDir() {
-#ifndef WIN32
-	return PACKAGE_LOCALE_DIR;
-#else
-	char exepath[MAX_PATH];
-	GetModuleFileName(NULL, exepath, MAX_PATH);
-	string datadir(exepath);
-	datadir.resize(datadir.find_last_of('\\'));
-	if (datadir.substr(datadir.length() - 4) == "\\bin" || datadir.substr(datadir.length() - 6) == "\\.libs") {
-		datadir.resize(datadir.find_last_of('\\'));
-		return datadir + "\\share\\locale";
-	}
-	return datadir + "\\locale";
-#endif
+	return getPackageDataDir() + "/locale";
 }
 
 string buildPath(string dir, string filename) {
